File: leetcode/495/c/main.c
#include <stdlib.h>

#define max(a, b) ((a) > (b) ? (a) : (b))

int findPoisonedDuration(int* timeSeries, int timeSeriesSize, int duration) {
    if (timeSeriesSize == 1) {
        return duration;
    }

    int elapsed = duration;
    int last_time = timeSeries[0] + duration;

    for (int i = 1; i < timeSeriesSize; i++) {
        int t = timeSeries[i];
        int next_last_time = t + duration;
        if (next_last_time <= last_time) {
            continue;
        }
        elapsed += next_last_time - max(last_time, t);
        last_time = next_last_time;
    }

    return elapsed;
}

File: leetcode/495/rs/src/main.rs
pub struct Solution;
impl Solution {
    pub fn find_poisoned_duration(time_series: Vec<i32>, duration: i32) -> i32 {
        if time_series.len() == 1 {
            return duration;
        }
        let mut elapsed = duration;
        let mut last_time = time_series[0] + duration;
        for &t in time_series.iter().skip(1) {
            let new_last_time = t + duration;
            if new_last_time <= last_time {
                continue;
            }

            elapsed += new_last_time - t.max(last_time);
            last_time = new_last_time;
        }

        elapsed
    }
}
fn main() {
    println!("Hello, world!");
}
